readthedocs/notifications/messages.py

```diff
@@ -0,0 +1,389 @@
+from django.utils.translation import gettext_noop as _
+
+from readthedocs.doc_builder.exceptions import (
+    BuildAppError,
+    BuildCancelled,
+    BuildMaxConcurrencyError,
+    BuildUserError,
+    MkDocsYAMLParseError,
+)
+from readthedocs.projects.constants import BUILD_COMMANDS_OUTPUT_PATH_HTML
+
+from .constants import ERROR, INFO, NOTE, TIP, WARNING
+
+
+class Message:
+    def __init__(self, id, header, body, type, icon_classes=None):
+        self.id = id
+        self.header = header
+        self.body = body
+        self.type = type  # (ERROR, WARNING, INFO, NOTE, TIP)
+        self.icon_classes = icon_classes
+        self.format_values = {}
+
+    def __repr__(self):
+        return f"<Message: {self.id}>"
+
+    def __str__(self):
+        return f"Message: {self.id} | {self.header}"
+
+    def set_format_values(self, format_values):
+        self.format_values = format_values or {}
+
+    def get_display_icon_classes(self):
+        if self.icon_classes:
+            return self.icon_classes
+
+        # Default classes that apply to all the notifications
+        classes = [
+            "fas",
+            "fa-user-group",
+            "fa-swap-opacity",
+        ]
+
+        if self.type == ERROR:
+            classes.append("fa-exclamation")
+        if self.type == WARNING:
+            classes.append("fa-triangle-exclamation")
+        if self.type == INFO:
+            classes.append("fa-????")
+        if self.type == NOTE:
+            classes.append("fa-????")
+        if self.type == TIP:
+            classes.append("fa-????")
+
+        return " ".join(classes)
+
+    def get_rendered_header(self):
+        return self.header.format(**self.format_values)
+
+    def get_rendered_body(self):
+        return self.body.format(**self.format_values)
+
+
+# TODO: review the copy of these notifications/messages on PR review and adapt them.
+# Most of them are copied from what we had in `readthedocs.doc_builder.exceptions`
+# and slightly adapted to have a header and a better body.
+BUILD_MESSAGES = [
+    Message(
+        id=BuildAppError.GENERIC_WITH_BUILD_ID,
+        header=_("Unknown problem"),
+        # Note the message receives the instance it's attached to
+        # and could be use it to inject related data
+        body=_(
+            """
+                There was a problem with Read the Docs while building your documentation.
+                Please try again later.
+                If this problem persists,
+                report this error to us with your build id ({instance.pk}).
+                """
```

Comment:
I suppose yes. The API response returns `\n    \n    Please try....` but that's not an issue because it's HTML. However, _I don't want that if possible_. I tried using `textwrap.wrap` and similar and I didn't find a good combination that removes all the trailing spaces and extra indentation.

Ideally, I'd like to:

- keep a clear indentation to make it easy to read the code
- don't use a big hundred chars "single-line" string
- don't use escape chars like `\` inside the string
- call the magic method (e.g. `textwrap.wrap`) only once inside the `registry.add()` method

Do you know a good way to do this?

Reply:
I opened https://github.com/readthedocs/readthedocs.org/issues/10987 to continue the discussion there.

Reply:
Mentioned in #10987 already, I was going to suggest `dedent`. This does require wrapping each string still though. There may be some gettext internals that could be pieced together, but that seems hard and being redundant is explicit and easy.

We can continue on #10987 though, I don't think this needs to hold up merging. However, it might be best to avoid altering our translation files until these strings are resolved. I don't quite know how gettext handles moved strings vs _moved and altered_ strings, but I suspect the additional whitespace could throw off all the existing translations.